File: Question 0049 - Group Anagrams/src/Solution.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

class Solution {
    public List<List<String>> groupAnagrams(String[] strs)
    {
        HashMap<String,List<String>> map = new HashMap<>();

        for (int i = 0; i < strs.length; i++) {
            char[] letters = strs[i].toCharArray();
            Arrays.sort(letters);

            String sortedWords = new String(letters);

            if(!map.containsKey(sortedWords)) map.put(sortedWords, new ArrayList<>());
            map.get(sortedWords).add(strs[i]);
        }
        return new ArrayList<>(map.values());
    }

}

File: Question 0049 - Group Anagrams/src/Solution.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution 
{
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) 
    {
        map<string, vector<string>> map;
        for(auto s : strs)
        {
            string letters = s;
            sort(letters.begin(), letters.end());
            map[letters].push_back(s);
        }

        vector<vector<string>> ans;
        for(auto& val : map)
        {
            ans.push_back(val.second);
        }

        return ans;
    }
};